src/index.ts

```diff
@@ -22,38 +23,92 @@ class Fetcher {
   #fetched: Set<string> = new Set()
   #queue: Queue
 
+  #userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.3"
+
   constructor(public options: Options) {
     const concurrency = options.concurrency || 3
     this.#queue = new Queue({ concurrency })
   }
 
-  #limitReached() {
-    return this.options.limit && this.#pages.size >= this.options.limit
-  }
-
-  #getContentSelector(pathname: string) {
-    if (typeof this.options.contentSelector === "function")
-      return this.options.contentSelector({ pathname })
-
-    return this.options.contentSelector
-  }
-
   async fetchSite(url: string) {
     logger.info(
       `Started fetching ${c.green(url)} with a concurrency of ${
         this.#queue.concurrency
       }`
     )
 
+    if (this.options.enableSitemap) {
+      logger.info(`Crawling sitemap...`)
+      await this.#crawlSitemap(url)
+    }
+
     await this.#fetchPage(url, {
```

Comment:
we probably don't need to continue normal fetching if sitemap is enabled, ie:

```ts
if (this.options.enableSitemap) {
 await this.#crawlSitemap(url)
} else {
  await this.#fetchPage(url, {})
}
```

Reply:
So, while sitemap.xml fetching is a great thing to have for larger sites or search-optimized sites, a lot of sites don't have it still. 
I was viewing the `enableSitemap` options as "we are enabling sitemap-based crawling if it's available" but also assuming that if someone calls `siteFetch()` they want the site fetched regardless of whether or not there is a sitemap available. E.g. if someone is batch-fetching a bunch of sites, they probably want the sites fetched and to use sitemap if possible, but don't want the site skipped just because there's not a sitemap.